### PR TITLE
Include clinical ID in the clinical attribute chart/column tooltip

### PR DESCRIPTION
**Pull request by zhx828**
Also do not show the description when the description is the same as dispaly name

# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/6354

Screen Shot 2019-07-12 at 11 17 54 AM
Screen Shot 2019-07-31 at 5 51 46 PM

### CODE REVIEW

**Comment by alisman** [comment]
@zhx828 the style of the tooltips in screenshots above are not totally consistent.  Is that because of scaling of the screenshots?

**Comment by zhx828** [comment]
@alisman I believe so. There is no special treatment there.

File: src/pages/studyView/StudyViewUtils.tsx [hunk 1]
```diff
@@ -208,10 +209,16 @@ const OPERATOR_MAP: {[op:string]: string} = {
...
 };
 
-export function getClinicalAttributeOverlay(displayName: string, description: string): JSX.Element {
-    return <div style={{maxWidth: '500px'}}>
-        <b>{displayName}</b><br/>
-        {description}
+export function getClinicalAttributeOverlay(displayName: string, description: string, clinicalAttributeId?: string): JSX.Element {
+    const comparisonDisplayName = displayName.toLowerCase().trim();
+    let comparisonDescription = description.toLowerCase().trim();
+    if (comparisonDescription.substring(comparisonDescription.length - 1) === '.') {
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx]: you could just do comparisonDescription.replace(/\.$/,"")   and you wouldn't have to bother checking for presence of dot at end.

File: src/pages/studyView/StudyViewUtils.tsx [hunk 2]
```diff
@@ -208,10 +209,16 @@ const OPERATOR_MAP: {[op:string]: string} = {
...
+        comparisonDescription = comparisonDescription.substring(0, comparisonDescription.length - 1);
+    }
+    return <div style={{maxWidth: '500px', display: 'flex', flexDirection: 'column'}}>
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx]: i'm not sure you need this flex stuff.  where you want it to break line, just use div and instead of span

**Comment by alisman** [comment]
in the second screenshot, is the tooltip meant to point to info icon?  is it misaligned?

**Comment by alisman** [comment]
also, seems like there should be a condition where we don't show the info button at all, no?

**Comment by zhx828** [comment]
> also, seems like there should be a condition where we don't show the info button at all, no?

@alisman we actually have a condition in the ChartHeader. But I need to update that logic now.

**Comment by zhx828** [comment]
> in the second screenshot, is the tooltip meant to point to info icon? is it misaligned?

@alisman I tried to adjust the tooltip placement and updated the screenshots. But it seems still not perfectly aligned.

**Pull request merged**